src/bundle/Controller/RoleController.php

```diff
@@ -188,6 +196,48 @@ public function createAction(Request $request): Response
         ]);
     }
 
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param \eZ\Publish\API\Repository\Values\User\Role $role
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function copyAction(Request $request, Role $role): Response
+    {
+        $this->denyAccessUnlessGranted(new Attribute('role', 'create'));
```

Comment:
Empty line between this line and below. You need to group statements so it's easier to read. Checking permissions is different thing from preparing a form so it makes sense to separate them.